mininext/topo.py

```diff
@@ -17,15 +17,16 @@ def __init__(self, nopts=None, **opts):
         BaseTopo.__init__(self, **opts)
 
     # Override addHost so that constructor defaults to MiniNExT host
-    def addHost(self, name, cls=Host, **opts):
+    def addHost(self, name, **opts):
         """Adds a host using the MiniNExT host constructor.
            name: host name
            cls: host constructor
            opts: host options
            returns: host name"""
         if not opts and self.hopts:
             opts = self.hopts
-        return BaseTopo.addNode(self, name, cls=cls, **opts)
+        opts['cls'] = Host
```

Comment:
The disadvantage of this is that it prevents further extensibility -- if you call `addHost('name', cls=NextGenHost`, `NextGenHost` will be overridden by `Host`. Can you either check if cls is set and not override, or just leave it in the constructor?